src/main/java/org/dice/factcheckbenchmark/benchmark/TaskGenerator.java

```diff
@@ -0,0 +1,62 @@
+package org.dice.factcheckbenchmark.benchmark;
+
+import org.hobbit.core.components.AbstractSequencingTaskGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+
+public class TaskGenerator extends AbstractSequencingTaskGenerator {
+    private static final Logger logger = LoggerFactory.getLogger(TaskGenerator.class);
+
+    @Override
+    public void init() throws Exception {
+        // Always init the super class first!
+        super.init();
+        logger.debug("Init()");
+        setAckTimeout(150000L);
+        // Your initialization code comes here...
+    }
+
+    @Override
+    protected void generateTask(byte[] data) throws Exception {
+
+        // Create tasks based on the incoming data inside this method.
+        // You might want to use the id of this task generator and the
+        // number of all task generators running in parallel.
+        logger.debug("generateTask()");
+
+        //TODO Research how these data members can be used
+        int dataGeneratorId = getGeneratorId();
+        int numberOfGenerators = getNumberOfGenerators();
+
+        // Create an ID for the task
+        String taskId = getNextTaskId();
+
+        //Split data using separator to extract query and expected
+        final String REGEX_SEPARATOR = ":\\*:";
+        String[] dataString = new String(data).split(REGEX_SEPARATOR);
+        dataString[1] = dataString[1].replaceAll("task/dataset/s", taskId + "/dataset/" + dataString[0]);
+
+        // Send the task to the system (and store the timestamp)
+        long timestamp = System.currentTimeMillis();
+        taskId = taskId + ":*:" + dataString[0];
```

Comment:
There was no need to do that. But the idea to implement this was to keep track of the files that performed poorly after the benchmark by making use of logs and looking at taskId. This could possibly help in future to determine if results are not good for specific context based patterns. Should I keep this implementation or remove it?

Reply:
Please remove it since 
1. it seems to increase the complexity of the API between benchmark and system (especially because the system adapter was parsing it as well)
2. it gives the system the name of the file that you loaded. Based on that, it might be able to cheat (i.e., to figure out whether the fact is true or false. Please keep in mind that we want to benchmark other systems as well :wink: )

It would be better if you simply log that a certain taskId uses file XYZ. The system adapter could log the taskId. With both logs you can still figure out which files are creating the issues.